### PR TITLE
planex-clone: new cli and preliminary refactoring

### PR DESCRIPTION
**Pull request by mseri**
This leaves the repatched flow unimplemented, will come in a separate PR

### CODE REVIEW

**Comment by mseri** [comment]
Travis is failing in a weird way on a code path that has not even been modified for a long time:
```
[Errno 13] Permission denied: '/var/lib/rpm/Basenames'
Traceback (most recent call last):
  File "/bin/planex-create-mock-config", line 9, in <module>
    load_entry_point('planex==4.0.0-beta4', 'console_scripts', 'planex-create-mock-config')()
  File "/usr/lib/python2.7/site-packages/planex-4.0.0_beta4-py2.7.egg/planex/cmd/createmockconfig.py", line 206, in main
    update_mock_repos(mock_repos, yum_repos, args.yum_config_opt)
  File "/usr/lib/python2.7/site-packages/planex-4.0.0_beta4-py2.7.egg/planex/cmd/createmockconfig.py", line 123, in update_mock_repos
    config.set(repo.id, 'mirrorlist', repo.mirrorlist)
  File "/usr/lib64/python2.7/ConfigParser.py", line 743, in set
    raise TypeError("option values must be strings")
TypeError: option values must be strings
```
It's as if the yum config in Travis is broken

File: planex/cmd/clone.py [hunk 1]
```diff
@@ -122,6 +141,109 @@ def apply_patchqueue(base_repo, pq_repo, pq_dir):
...
+                 "a git repository.")
+    if 0 not in patchqueues:
+        sys.exit("error: planex-clone requires PatchQueue0 to point "
+                 "to a git repository.")
+
+    if len(patchqueues.keys()) > 1:
+        sys.exit(
+            "error: planex-clone does not support the cloning and "
+            "assembly of multiple sources and patchqueues, currently "
+            "this case needs to be handled by hands.")
```
Comment (srowe) [inline planex/cmd/clone.py]: "... handled manually."

File: planex/cmd/clone.py [hunk 2]
```diff
@@ -122,6 +141,109 @@ def apply_patchqueue(base_repo, pq_repo, pq_dir):
...
+        print(gce.stderr)
+
+
+def assemble_repatched(args, pin):
+    """
+    Assemble an upstream centos package and apply the upstream patchqueue,
+    then apply the local patchqueue.
+    """
+    print(args, pin)
+    raise Exception("Unimplemented")
```
Comment (srowe) [inline planex/cmd/clone.py]: Use the standard exception `NotImplementedError`

Comment (mseri) [reply planex/cmd/clone.py]: Thanks!!! I could not remember or find it

**Comment by mseri** [comment]
I have squashed the fixups. I'll see if I can figure out why epel is broken on travis

**Pull request merged**